packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -464,19 +465,78 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
             const content = response.content;
+            let isResourceDataAvailable = false;
+            let dnstime; // domainLookupEnd - domainLookupStart
+            let redirectTime; // redirectEnd -redirectStart
+            let tcpHandshakeTime; // connectEnd  - connectStart
+            let secureConntime; // connectEnd  - secureConnectionStart
+            let responseTime; // responsEnd - responseStart
+            let fetchStToRespEndTime; // responseEnd  - fetchStart
+            let reqStToRespEndTime; // responseEnd - requestStart
+            let stToRespEnd; // responseEnd - startTime
+            const spReqDuration = response.headers.get("sprequestduration");
+
+            if (performance) {
+                const resources1 = performance.getEntriesByType("resource");
+                for (let i = resources1.length - 1; i > 0; i--) {
+                    const indResTime = resources1[i] as PerformanceResourceTiming;
+                    const resource_name = indResTime.name;
+                    const resource_initiatortype = indResTime.initiatorType;
+                    if ((resource_initiatortype.localeCompare("fetch") === 0) && (resource_name.localeCompare(url) === 0)) {
+                        isResourceDataAvailable = true;
+                        redirectTime = indResTime.redirectEnd - indResTime.redirectStart;
+                        dnstime = indResTime.domainLookupEnd - indResTime.domainLookupStart;
+                        tcpHandshakeTime = indResTime.connectEnd - indResTime.connectStart;
+                        secureConntime = (indResTime.secureConnectionStart > 0) ? (indResTime.connectEnd - indResTime.secureConnectionStart) : "0";
+                        responseTime = indResTime.responseEnd - indResTime.responseStart;
+                        fetchStToRespEndTime = (indResTime.fetchStart > 0) ? (indResTime.responseEnd - indResTime.fetchStart) : "0";
+                        reqStToRespEndTime = (indResTime.requestStart > 0) ? (indResTime.responseEnd - indResTime.requestStart) : "0";
+                        stToRespEnd = (indResTime.startTime > 0) ? (indResTime.responseEnd - indResTime.startTime) : "0";
+                        break;
+                    } else {
+                        isResourceDataAvailable = false;
+                    }
+                }
+            }
+            if (isResourceDataAvailable === false) {
+                redirectTime = -1;
+                dnstime = -1;
+                tcpHandshakeTime = -1;
+                secureConntime = -1;
+                responseTime = -1;
+                fetchStToRespEndTime = -1;
+                reqStToRespEndTime = -1;
+                stToRespEnd = -1;
+            }
+            let clientDuration;
+            if (spReqDuration == null) {
+                    clientDuration = -1;
+            } else {
+                    clientDuration = parseInt(spReqDuration, 10) - responseTime;
```

Comment:
We need a local timer started before the fetchHelper method is called above. Use that local timer to track start-end, and compute overallTime = end-start, serviceTime = spReqDuration (you have it already), networkTime = lastResourceByteTime - resources[i].startTime, clientTime = overallTime - serviceTime - networkTime.

Reply:
Made changes to get clienttime, pl. take a look in new output and code changes.

Reply:
Looks good - dont need the clientDuration field any more?

Reply:
agreed, removed it.